landscaper/cmd/landscaper-gardenlet/app/landscaper-gardenlet.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	gardenlet "github.com/gardener/gardener/landscaper/gardenlet/pkg"
+	"github.com/gardener/gardener/landscaper/gardenlet/pkg/apis/imports"
+	importsv1alpha1 "github.com/gardener/gardener/landscaper/gardenlet/pkg/apis/imports/v1alpha1"
+	importvalidation "github.com/gardener/gardener/landscaper/gardenlet/pkg/apis/imports/validation"
+
+	gardenletconfig "github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+	"k8s.io/component-base/version/verflag"
+	landscaperconstants "github.com/gardener/landscaper/apis/deployer/container"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+
+// in integration tests, we do not assume that the Gardenlet can be rolled out successfully,
+// nor that the Seed can be registered.
+// this is to provide an easy means of testing the landscaper component without requiring
+// a fully functional Gardener control plane.
+var isIntegrationTest bool
```

Comment:
What about renaming this to `waitForGardenletRollout` and removing the comments talking about test cases?

Reply:
It is not only about waiting for Gardenlet rollout but also not checking if the Seed already exists and that the Seed is registered correctly.
It is something I like to get rid of as soon at is possible from the TM side. 
Something like _doNotAssumeGardenerAPI_ seems more fitting but I am also happy with this temporary. flag.

Reply:
Yeah, sure, if you have a better name then use it, fine for me. I just wanted to propose to have flag names that clearly indicate what happens if you use it (`isIntegrationTest` doesn't tell me anything and I would need to lookup the help).

Reply:
Can we just leave the flag name like it is? 
The flag should never be used outside of integration test and is a temporary solution that will be removed asap.
If someone wants to use it, it is well documented in multiple places telling them that it is only for integration tests.

Calling it something like _NoGardenerAPIs_ etc. seems awkward to me and would raise even more questions for me.